src/ol_openedx_git_auto_export/ol_openedx_git_auto_export/signals.py

```diff
@@ -19,32 +31,99 @@ def listen_for_course_publish(
     """
     Receives publishing signal and performs publishing related workflows
     """
+
     if settings.FEATURES.get("ENABLE_EXPORT_GIT") and settings.FEATURES.get(
         ENABLE_GIT_AUTO_EXPORT
     ):
         course_module = modulestore().get_course(course_key)
-        if not course_module.giturl:
-            log.info(
-                "Course %s does not have a GIT URL set in course advance settings, skipping export.",  # noqa: E501
-                course_module.id,
-            )
-            return
-
-        git_repo_export_dir = getattr(
-            settings, "GIT_REPO_EXPORT_DIR", "/openedx/export_course_repos"
-        )
-        if not os.path.exists(git_repo_export_dir):  # noqa: PTH110
-            # for development/docker/vagrant if GIT_REPO_EXPORT_DIR folder does not exist then create it  # noqa: E501
-            log.error(
-                "GIT_REPO_EXPORT_DIR is not available in settings, please create it first"  # noqa: E501
-            )
-            os.makedirs(git_repo_export_dir, 0o755)  # noqa: PTH103
-
+        get_or_create_git_export_repo_dir()
         log.info(
             "Course published with auto-export enabled. Starting export... (course id: %s)",  # noqa: E501
             course_key,
         )
+        # HACK: To create auto git repo for Re-runs as it does not emit COURSE_CREATED signal  # noqa: E501 FIX004
```

Comment:
Is it a bug that reruns don't emit `COURSE_CREATED` signal?

Is there a different signal emitted when a rerun is created? @asadali145

Reply:
> Is there a different signal emitted when a rerun is created? @asadali145

I am not sure about a specific signal, but we listen for the post_save signal on changes in `CourseRerunState` model. It actually tracks the status of reruns, like in_progress, failed, and completed. When a rerun state is completed, then we update our course sync mappings.

Reply:
thanks @asadali145 I will update it to catch course re-run creation through `post_save` signal 👍